src/librustc_trans/back/write.rs

```diff
@@ -613,13 +622,73 @@ unsafe fn optimize_and_codegen(cgcx: &CodegenContext,
         }
     }
 
+    if config.obj_needs_as {
+        // XXX most of the logic here has been copied from the link_natively
+        // function (src/librustc_trans/back/link.rs)
+        // TODO don't hardcode, maybe expose as a `as` field in the target
+        // specification
+        // TODO how to properly access `sess` here?
+        let mut cmd = Command::new("msp430-as");
```

Comment:
(note that this block of additions I think can be subsumed by `write::run_assembler`)